lfs/credentials.go

```diff
@@ -87,6 +97,27 @@ func getCredURLForAPI(req *http.Request) (*url.URL, error) {
 	return credsUrl, nil
 }
 
+func setCredURLFromNetrc(req *http.Request) bool {
+	home := os.Getenv("HOME")
+	if len(home) == 0 {
+		return false
+	}
+
+	nrc := filepath.Join(home, ".netrc")
```

Comment:
Do you think it's worth caching this in the global `lfs.Config` object? Looks like this is called for every API request, so it could be used multiple times in a single `git-lfs` process (especially with the legacy API).